addons/addon-base-raas/packages/base-raas-services/lib/data-source/data-source-registration-service.js

```diff
@@ -79,6 +87,150 @@ class DataSourceRegistrationService extends Service {
+      { workflowId: workflowIds.bulkCheck },
+      {
+        status,
+      },
+    );
+    // Write audit event
+    await this.audit(requestContext, {
+      action: 'bulk-check-reachability',
+      body: { status },
+    });
+  }
+
+  async reachDsAccount(requestContext, { id, type }) {
+    const accountService = await this.service('dataSourceAccountService');
+    const dataSourceAccount = await accountService.mustFind(requestContext, { id });
+
+    if (!dataSourceAccount.status) {
+      throw this.boom.badRequest('Can only check reachability for data source account', true);
+    }
+    const prevStatus = dataSourceAccount.status;
+    let newStatus = prevStatus;
+    let statusMsg = '';
+
+    // TODO: Use the cloudformation stack name to access its status in the data source account
+    // TODO: Determine if the stack is compliant
+    const reachable = false;
+
+    if (reachable) {
+      newStatus = 'reachable';
+      if (prevStatus !== newStatus) {
+        const workflowTriggerService = await this.service('workflowTriggerService');
+        await workflowTriggerService.triggerWorkflow(
+          requestContext,
+          { workflowId: workflowIds.accountStatusChange },
+          {
+            id,
+            type,
+          },
+        );
+      }
+    } else if (prevStatus === 'pending') {
+      statusMsg = `WARN|||Data source account ${id} is not reachable yet`;
+    } else {
+      statusMsg = `ERR|||Error getting information from data source account ${id}`;
+    }
+    await accountService.updateStatus(requestContext, dataSourceAccount, { status: newStatus, statusMsg });
+    const outputVal = newStatus;
+
+    // Write audit event
+    await this.audit(requestContext, {
+      action: 'check-dsAccount-reachability',
+      body: { id, type },
+    });
+
+    return outputVal;
+  }
+
+  async reachStudy(requestContext, { id, type }) {
+    const studyService = await this.service('studyService');
+    const studyEntity = await studyService.mustFind(requestContext, id);
+
+    if (!studyEntity.status) {
+      throw this.boom.badRequest('Can only check reachability for data source account studies', true);
+    }
+
+    const prevStatus = studyEntity.status;
+    let newStatus = prevStatus;
+    let statusMsg = '';
+    const reachable = await this._assumeAppRole(requestContext, studyEntity.appRoleArn);
+
+    if (reachable) {
+      newStatus = 'reachable';
+    } else if (prevStatus === 'pending') {
+      statusMsg = `WARN|||Study ${id} is not reachable yet`;
+    } else {
+      statusMsg = `ERR|||Error getting information from study ${id}`;
+    }
+    await studyService.updateStatus(requestContext, studyEntity, { status: newStatus, statusMsg });
+    const outputVal = newStatus;
+
+    // Write audit event
+    await this.audit(requestContext, {
+      action: 'check-study-reachability',
+      body: { id, type },
+    });
+
+    return outputVal;
+  }
+
+  async _assumeAppRole(requestContext, appRoleArn) {
```

Comment:
Also, I made it so that, we can assume the application role to check if the study is readable, can you add that logic? so you can do something like
`const s3Client = await aws.getClientSdkForRole({ roleArn, appRoleArn, clientName: 'S3' }); 
// then use s3Client to read the head of an object`

Reply:
Done.